spyder/widgets/variableexplorer/dataframeeditor.py

```diff
@@ -197,6 +198,10 @@ def headerData(self, section, orientation, role=Qt.DisplayRole):
                 except:
                     header = to_text_string(self.df_header[0])
                 return to_qvariant(header)
+            elif PY2 and isinstance(self.df_header[section-1], TEXT_TYPES):
+                header = self.df_header[section-1]

```

Comment:
Please add a comment above this line explaining the need of this change, like in the previous `elif`.